### PR TITLE
add gtid set information in xid event

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
add gtid set information in xid event

### CODE REVIEW

File: replication/event.go [hunk 1]
```diff
@@ -216,6 +216,9 @@ func (e *RotateEvent) Dump(w io.Writer) {
 
 type XIDEvent struct {
 	XID uint64
+
+	// in fact XIDEvent dosen't have the GTIDSet information, just for beneficial to use
+	gset GTIDSet
```
Comment (GregoryIan) [inline replication/event.go]: let `gset` be a public property

Comment (siddontang) [inline replication/event.go]: +1

**Comment by WangXiangUSTC** [comment]
@siddontang @GregoryIan I had changed gset to GSet

**Comment by siddontang** [comment]
LGTM

PTAL @GregoryIan

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**